spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Log.java

```diff
@@ -40,7 +43,12 @@
 	 */
 	private final String event;
 
-	public Log(long timestamp, String event) {
+	@JsonCreator
+	public Log(
+			@JsonProperty(value = "timestamp", required = true) long timestamp,
```

Comment:
Nah I definitely prefer the constructor injection. Initially I did the same but for some reason it failed